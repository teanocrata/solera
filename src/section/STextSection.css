@import url('https://fonts.googleapis.com/css?family=Gloria+Hallelujah');

h2 {
  font-family: "Gloria Hallelujah", cursive;
  color: forestgreen;
  max-width: 85%;
  text-align: center;
  font-size: 80px;
  text-transform: none;
}

p {
  text-align: center;
}

@media (max-width: 768px) {
  h2 {
    font-size: 50px;
  }
}

@media (max-width: 425px) {
  h2 {
    font-size: 35px;
  }
}
