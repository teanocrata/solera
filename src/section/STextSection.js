import React, { Component } from "react";
import { Grid, Row, Col } from "react-bootstrap";
import "./STextSection.css";

class STextSection extends Component {
  render() {
    return this.props.text ? (
      <Grid fluid>
        <Row
          className="show-grid"
          style={{
            backgroundColor: "white"
          }}
        >
          <Col md={12} className="slideanim">
            <h2>{this.props.text.title}</h2>
            <br />
            {this.props.text.pharagraphs ? (
              this.props.text.pharagraphs.map(
                (pharagraph, index) =>
                  pharagraph.size === "small" ? (
                    <p key={index}>
                      <small>{pharagraph.text}</small>
                    </p>
                  ) : (
                    <p key={index}>{pharagraph.text}</p>
                  )
              )
            ) : (
              ""
            )}
          </Col>
        </Row>
      </Grid>
    ) : null;
  }
}

export default STextSection;
