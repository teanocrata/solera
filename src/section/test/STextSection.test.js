import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import STextSection from "../STextSection";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<STextSection />, div);
});

test("STextSection renders correctly", () => {
  const section = renderer.create(<STextSection />).toJSON();
  expect(section).toMatchSnapshot();
});
