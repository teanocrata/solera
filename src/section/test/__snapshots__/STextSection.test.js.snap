// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STextSection renders correctly 1`] = `null`;

exports[`STextSection renders correctly with complete text data 1`] = `
<div
  className="container-fluid"
>
  <div
    className="show-grid row"
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <div
      className="slideanim col-md-12"
    >
      <h2>
        Bienvenidos
      </h2>
      <br />
      <p>
        Visita nuestra escuela y disfruta de una agradable sensación de calma y tranquilidad en el campo.
      </p>
      <p>
        Verás cómo viven nuestros caballos, conocerás los diferentes trabajos que hacemos con ellos, las distintas clases de domas y por qué nosotros optamos por la doma natural y muchas más curiosidades..
      </p>
      <p>
        <small>
          ¡Y podrás probar lo que significa conectar con un animal!
        </small>
      </p>
    </div>
  </div>
</div>
`;

exports[`STextSection renders correctly with text data only with title 1`] = `
<div
  className="container-fluid"
>
  <div
    className="show-grid row"
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <div
      className="slideanim col-md-12"
    >
      <h2>
        Bienvenidos
      </h2>
      <br />
      
    </div>
  </div>
</div>
`;

exports[`STextSection renders correctly with text data without pharagraphs 1`] = `
<div
  className="container-fluid"
>
  <div
    className="show-grid row"
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <div
      className="slideanim col-md-12"
    >
      <h2>
        Bienvenidos
      </h2>
      <br />
    </div>
  </div>
</div>
`;

exports[`STextSection renders correctly with text data without pharagraphs and title 1`] = `
<div
  className="container-fluid"
>
  <div
    className="show-grid row"
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <div
      className="slideanim col-md-12"
    >
      <h2 />
      <br />
    </div>
  </div>
</div>
`;
